src/sentry_value.c

```diff
@@ -308,7 +308,11 @@ sentry_value_new_list(void)
     list_t *l = SENTRY_MAKE(list_t);
     if (l) {
         memset(l, 0, sizeof(list_t));
-        return new_thing_value(l, THING_TYPE_LIST);
+        sentry_value_t rv = new_thing_value(l, THING_TYPE_LIST);
+        if (sentry_value_is_null(rv)) {
```

Comment:
These cases here are interesting. Sure, `new_thing_value` allocates internally and can fail. But `CTU` can’t look through the `sentry_value_is_null` assertion and will still flag all of these as false positives.

Reply:
The problem why I do all the checks outside is that `new_thing_value` takes ownership of whatever its wrapping, but it doesn’t know how to free. Maybe with a separate function pointer argument, this becomes clearer.

Reply:
Usually, there are attributes you can put on the function to declare this behavior. Since we encode the pointers in a strange way, it's understandable that tools don't follow the path. 

Let's skip this for now.